Convert dependency check annotation from `h3` to `p`

mokagio opened the pull request:
An `h3` (`###` markdown) in the CI annotation results in a message that clashes visually with the rest of the annotations.

See for example https://buildkite.com/automattic/simplenote-ios/builds/1146

**Before**
![image](https://github.com/user-attachments/assets/9886a037-e277-4b22-b4de-e4fffb35b510)


**After** via editing the HTML source in the page. No build run with this yet.
![image](https://github.com/user-attachments/assets/24a15090-9a32-4119-ba4e-bb870bb1505f)


### Test
We'll see this in action on the next code freeze.

### Review
Only one infrastructure developer required to review these changes, but anyone can perform the review.

### Release

These changes do not require release notes.

mokagio committed:
Remove `h3` header from pods check step during code freeze

An `h3` (`###` markdown) in the CI annotation results in a message that
clashes visually with the rest of the annotations.

See for example https://buildkite.com/automattic/simplenote-ios/builds/1146

mokagio committed:
Reintroduce check pods annotation header, but as a paragraph

mokagio committed:
Tidy: Extract compound condition to variable for readability

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio commented on fastlane/lanes/release.rb:
```diff
@@ -312,9 +312,13 @@ def check_pods_references
-  buildkite_annotate(context: 'pods-check', style: style, message: message) if is_ci
+  return unless is_ci
+
+  all_dependencies_stable = result[:pods].nil? || result[:pods].empty?
```
I did this tidy because I thought I'd add an emoji an wanted to DRY the check.

I decided not to add the emoji to the text because Buildkite already has icons and colors for each annotation style. But I kept the code like this because I think it's a tiny readability improvements.

Also, since I'm on my way to write more commentary than code, I considered being explicit in the variable names and the message in mentioning CocoaPods. But, I'd say leaving it as generic "dependencies" makes it easier for us to update the logic to incorporate something that parses the `Package.resolved` JSON looking for dependencies that point to non stable versions.

mokagio commented on fastlane/lanes/release.rb:
```diff
@@ -312,9 +312,13 @@ def check_pods_references
+  all_dependencies_stable = result[:pods].nil? || result[:pods].empty?
+  style = all_dependencies_stable ? 'success' : 'warning'
+  # Notice the double new line to produce a new HTML paragraph from the Markdown formatting.
+  message = "Internal dependencies check result:\n\n#{result[:message]}"
```
```suggestion
  message = "Stable dependencies check result:\n\n#{result[:message]}"
```

mokagio committed:
Replace "Internal" with "Stable"

The `ios_check_beta_deps` action doesn't discriminated between internal (1st party) and external (3rd party) dependencies.

https://github.com/wordpress-mobile/release-toolkit/blob/ee03eb9232597fd812c18d3fb0fbf91130facf97/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_check_beta_deps.rb

AliSoftware reviewed (approved):
WFM, but you'll have to update the wording of the task referencing that annotation in the ReleaseV2 scenario to match, now that the annotation has a different "title":

<img width="1229" alt="image" src="https://github.com/user-attachments/assets/5b3fa4f1-3076-4e1e-9681-2972cc7afeca">

mokagio commented:
>  but you'll have to update the wording of the task referencing that annotation in the ReleaseV2 scenario to match

Done, can be verified in the scenario for 4.56 and later.

mokagio committed:
Merge branch 'trunk' into mokagio/remove-pods-check-header

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/simplenote-ios/milestone/70">4.55</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>
